R/ComputeFirstDifferences.R

```diff
@@ -44,34 +44,33 @@ ComputeFirstDifferences <- function(df, idvar, timevar, diffvar,
         stop("return_df should be logical.")
     }
 
-    data.table::setDT(df)
-    data.table::setorderv(df, cols = c(idvar, timevar))
+    data.table::setorderv(dt, cols = c(idvar, timevar))
 
     if (!timevar_holes) {
-        df[, paste0(diffvar, "_fd") := get(diffvar) - shift((get(diffvar))),
+        dt[, paste0(diffvar, "_fd") := get(diffvar) - shift((get(diffvar))),
            by = idvar]
     } else {
         ## Create dataset with all combinations to compute first differences
-        all_combinations <- CJ(unique(df[[idvar]]),
-                               min(df[[timevar]]):max(df[[timevar]]))
+        all_combinations <- data.table::CJ(unique(dt[[idvar]]),
+                               min(dt[[timevar]]):max(dt[[timevar]]))
         setnames(all_combinations, new = c(idvar, timevar))
 
-        df_all <- merge(df, all_combinations,
+        dt_all <- data.table::merge.data.table(dt, all_combinations,
                         by = c(idvar, timevar), all = TRUE)
 
-        df_all[, paste0(diffvar, "_fd") := get(diffvar) - shift((get(diffvar))),
+        dt_all[, paste0(diffvar, "_fd") := get(diffvar) - shift((get(diffvar))),
                 by = idvar]
 
         ## Bring first differences back to the original dataset
         vars_to_keep <- c(idvar, timevar, paste0(diffvar, "_fd"))
 
-        df <- merge(df, df_all[, .SD, .SDcols = vars_to_keep],
+        dt <- data.table::merge.data.table(dt, dt_all[, .SD, .SDcols = vars_to_keep],
                     by = c(idvar, timevar), all.x = TRUE)
     }
 
     if (return_df) {
-        df <- as.data.frame(df)
+        return(as.data.frame(dt))
     }
```

Comment:
Should we remove this option?

Reply:
Thanks! Looking at da88346, I think that you forgot removing the argument in `R/ComputeFirstDifferences.R`? Can we check whether we have similar `return_df` arguments in other functions that we can delete?